### DISCUSSION TITLE
Расчёт распределяемых значений, которые потом нужны в других таблицах

### DISCUSSION THREAD

**Discussion by gorbarov** [totumonline/totum-ru-issues-and-discussions#200] [Как это сделать?] [answered]
И ещё вопрос, я пробовал что-то подобное, но у меня не получилось.

В отчётной таблице нужно распределить косвенные расходы на проекты, в зависимости от потраченных часов или выручки.
Но эти данные нужно сохранить, чтобы отобразить в свойствах каждого отдельного проекта.

Возможно, я мыслю сейчас как экселевод, но я не до конца понимаю, как это можно сделать (какой типа таблицы, как рассчитывать зависимые значения и тд).

Вот как это выглядит в экселе. Есть список проектов, у них выручка или часы есть. И есть расходы, которые нужно распределить пропорционально часам или выручке

![e327f805fe](https://user-images.githubusercontent.com/55755565/187025001-d5ca3f3f-d54b-4538-992a-a97ff6782886.png)

Там из формулы видно, в зависимости от того сколько часов было потрачено в проекте по отношению ко всем часам распределяются косвенные расходы.

Какую магию надо сделать, чтобы это заработало?

**Comment by erpic-ru** [comment, marked as answer]
Если данные по распределению косвенных расходов нужно где-то использовать, то следует делать это в расчетных таблицах в цикле. Т.е. такая таблица (цикл) создается, например, на каждый месяц и данные из них можно забирать куда угодно.

**Reply by gorbarov** [reply]
А как потом собрать из таблиц циклов данные в отчёт за год?

**Reply by gorbarov** [reply]
Сам спросил сам ответил https://www.youtube.com/watch?v=-Ls48xQ1Gx4&list=PLfiZJiSJ6dosChEUL2nd6DX4JCLQ2oA83&ab_channel=totumonlineRU

**Reply by ttm-alexeypolunin** [reply]
### Важно!

В скриншоте в вопросе: в **последующей** колонке считается доля от суммы **предыдущей** колонки. Что бы это работало внутри циклов надо переключить порядок расчет таблицы в **колоночный режим:** https://ru.docs.totum.online/calculation-sequence#table-full-column
